solana-common/src/instruction.rs

```diff
@@ -0,0 +1,67 @@
+use crate::pb::sol::instructions::v1::Instruction;
+use crate::pb::sol::instructions::v1::Instructions;
+use crate::pb::sol::transactions::v1::Transactions;
+use substreams::pb::sf::substreams::index::v1::Keys;
+use substreams::matches_keys_in_parsed_expr;
+use substreams_solana::pb::sf::solana::r#type::v1::ConfirmedTransaction;
+
+#[substreams::handlers::map]
+fn all_instructions_without_votes(transactions: Transactions) -> Result<Instructions, substreams::errors::Error> {
+    let instructions = get_instructions_from_transactions(&transactions.transactions);
+
+    Ok(Instructions {
+        instructions,
+    })
+}
+
+#[substreams::handlers::map]
+fn index_instructions_without_votes(instructions: Instructions) -> Result<Keys, substreams::errors::Error> {
+    let keys: Vec<String> = instructions.instructions
+            .into_iter()
+            .map(|inst| format!("program:{}", inst.program_id))
+            .collect();
+
+    Ok(Keys { keys })
+}
+
+#[substreams::handlers::map]
+fn filtered_instructions_without_votes(query: String, instructions: Instructions) -> Result<Instructions, substreams::errors::Error> {
+    let filtered_instructions = instructions.instructions
+            .into_iter()
+            .filter(|inst| {
+                let mut keys = Vec::new();
+                keys.push(format!("program:{}", inst.program_id));
+
+                matches_keys_in_parsed_expr(&keys, &query).expect("matching events from query")
+            }).collect();
+
+    Ok(Instructions {
+        instructions: filtered_instructions
+    })
+}
+
+pub fn get_instructions_from_transactions(transactions: &Vec<ConfirmedTransaction>) -> Vec<Instruction> {
+    let mut processed_instructions: Vec<Instruction> = Vec::new();
+
+    for confirmed_transaction in transactions.iter() {
+        let acct_keys: Vec<&[u8]> = confirmed_transaction.resolved_accounts().iter().map(|k| k.as_slice()).collect();
+
+        if let Some(transaction) = &confirmed_transaction.transaction {
+            for instruction in confirmed_transaction.all_instructions().into_iter() {
```

Comment:
```suggestion
            for instruction in confirmed_transaction.all_instructions().iter() {
```

Should be good.

Reply:
It seems the method is not available, only `into_iter()`

Reply:
Try `for instruction in confirmed_transaction.all_instructions() {` it should work